doc/sign-package.md

```diff
@@ -159,3 +159,95 @@ If the host arguments are not allowed to be passed then any specified within the
 If any host environment variables are configured as available within the SGX enclave, then this command will pass them though. Enclave specific environment variables will be added once Mystikos transfers control to the enclave.
 
 You then can ship your application in the form of this single exectuable to your desired platform.
+
+#### Packaging with EXT2 
+
+##### Signing EXT2 image
+
+Unlike packaging with a CPIO archive, an EXT2 image is not part of the signed package.
+Signing the EXT2 image allows us to ensure that Mystikos will only
+load a file system that matches the root hash or public key that we specified.
+
+> Signed EXT2 images are only supported for the root file system.
+Any additional file system mounted through configuration (auto-mount) or through
+calling `mount()` in the application (explicit mount) will not have the
+signature verification.
+
+
+The following command creates a signed EXT2 image.
+
+```
+$ myst mkext2 --sign=private.pem appdir ext2image
+```
+
+The ``private.pem`` file is the private key.
+
+The following generates test keys.
+
+```
+$ openssl genrsa -out private.pem -3 3072
+$ openssl rsa -in private.pem -pubout -out public.pem
+```
+
+##### Packaging with EXT2 image
+
+The EXT2 image created above can be loaded by a Mystikos application that
```

Comment:
`Mystikos application` -> `Mystikos runtime`

Reply:
`trust the given signing authority` -> `trust the app launching process`. The myst runtime trusts whatever `--roothash` or `--pubkey` from the command line, which is much less secure than an application packaged with roothash or pubkey. We should emphasize that. Actually, I'd just say packaging is recommended over signing for ext2fs, and remove paragraphs related to signing.

Reply:
Are you talking about `myst exec-sgx --roothash=roothash ..`? This section is about packaging (`myst package-sgx --roothash=roothash ...`). Signing is still necessary for the ext2 image, since it is not part of the packaged application image. If you do specify an `appdir` when you run `myst package-sgx`, you're actually including a CPIO image with the final binary.

Reply:
Line 165-189 are what I meant `paragraphs related to signing`.

Packaging implicitly calls signing. 

Yes, `myst exec-sgx --roothash=roothash ...` is what I meant `trust the app launching process`. Otherwise an attacker could 1) tamper app, 2) re-generate the roothash, and 3) launch it with the new roothash.

Reply:
`Mystikos application` was meant to refer to a signed package. I'll rephrase for clarity.